### ISSUE TITLE
python port

### ISSUE THREAD

**Issue by vors** [davechurchill/commandcenter#30]
Hello, David!

I'm curious about your thoughts on https://github.com/deepmind/pysc2 and an option to align commandcenter-based AI and the pysc2 training agents, for some interop maybe?
My interest in this question arises from the fact that I'd love to use awesome primitives you built in the commandcenter, but at the same time, I'd love to have the flexibility to plug in all the RL training stuff that deepmind team provided. Also doing it casually in Python would be somewhat easier than in c++ for me and possibly some other sc2 enthusiasts.

**Comment by davechurchill** [comment]
If someone does something involving pysc2 / commandcenter and wants to pull request what they did, I would be happy to review it, but I have no plans of doing anything but C++ with CommandCenter for the foreseeable future

**Comment by alkurbatov** [comment]
The problem is that there is no direct connection between the c++ code and the Python code. From my point of view the only thing that could be done here is creation of the Python bindings over Python API that will allow to use CommandCenter in Python apps. However it is a hard work to support such bindings in actul state.

**Comment by vors** [comment]
@alkurbatov yes, that's my understanding too. It would be doable if we can keep an API surface, but it's additional restrictions. I was more thinking along the lines of rewriting the whole project in python for the sake of interop.

**Comment by davechurchill** [comment]
Seems like a lot of work to avoid learning a little C++? :D

**Comment by vors** [comment]
lol sure

My hypothesis is that having 2 projects for the sc2 research that can reuse each other is better than having 2 projects that have no way to interact and can lead to more interesting results. But I don't have any data to prove it.

**Comment by davechurchill** [comment]
True - I just don't have an immediate use for it so I'm going to bow out of any responsibility for it :)

I suspected that using some sort of python / c++ interop may be the easiest way of going about it

**Comment by alkurbatov** [comment]
@vors I did such things for more complicated C-like API. We used handmade codegenerator to analyze C++ code and create Python bindings (each of which converted arguments and executed a C++ function). While the codegenerator allows to reduce support burden (i.e. just recompile everything when needed) on the other hand we have to support the codegenerator and expand it for some complex datatypes, and fix leaks in the generated code.

I am not sure if such approach is applicable here but it could be an option too.
